### DISCUSSION TITLE
Ability to use own Postgres db

### DISCUSSION THREAD

**Discussion by mariusa** [supabase/supabase#225] [Questions] [answered]
# Feature request

We have Postgres db already hosted somewhere else (Amazon, Heroku...)

Please let us use those instead of creating new ones, eg by providing a connection string.

Speed & security (eg some users prefer internal company databases be only inside company firewall) will be improved with 

> When we get out of alpha, we will release a fat middleware server which stitches together the the open source tools we are using.

so one can deploy supabase inside the company firewall.

This also means creating `auth` function in that db, maybe others? Will supabase do this automatically or provide a script to run?

**Comment by awalias** [comment, marked as answer]
the first version of supabase actually worked this way :) there was some difficulty with identifying when some config was missing in the db - e.g. logical replication not enabled  etc. but nothing that can't be solved

this is back on the roadmap but probably won't be completed within the next couple of months

re: the auth schema we should be able to just inject this automatically (which is what we do with the hosted dbs today)

**Comment by kiwicopple** [comment]
Thanks for this one @mariusa - I'm going to shift it over to a Discussion 💬
